raiden-dapp/src/components/dialogs/UdcDepositDialog.vue

```diff
@@ -135,7 +136,27 @@ export default class UdcDepositDialog extends Vue {
   }
 
   get valid(): boolean {
-    return /^[1-9]\d*$/.test(this.defaultUtilityTokenAmount);
+    let utilityTokenAmount: BigNumber;
+
+    try {
+      utilityTokenAmount = parseUnits(
+        this.defaultUtilityTokenAmount,
+        this.udcToken.decimals
+      );
+    } catch (err) {
+      utilityTokenAmount = Zero;
+    }
+
+    if (!this.mainnet) return utilityTokenAmount.gt(Zero);
```

Comment:
Nitpick: with the comment below (about taking out the balance constant), it'd be clearer here that the two conditions (on mainnet and not) would be in an `if/else` block. The way it is now, looks like the amount is only required to be positive if on testnet, and then the person needs to go to the other return to see the same condition is also checked there with something else (`amout <= balance`, since testnet can mint)

Reply:
I see, yes, added an `if/else` block.